### ISSUE TITLE
Set step and dropdown-step separatly

### ISSUE THREAD

**Issue by Springrbua** [vaadin/vaadin-time-picker#173]
In many cases, it is enough to set time in f.e. 15min steps. However, I would like to allow the users to manually enter more accurate values. Also, setting a step of less then 15min would disable the overlay and the user would always need to manually enter the value.
So it would be nice to be able to control the step and the steps for the dropdown separately.

**Comment by rolfsmeds** [comment]
Actually you can already do this: the step size in vaadin-time-picker only affects the dropdown and the arrow key increment/decrement, but does not prevent the user from entering any time value into the field manually.

**Comment by Springrbua** [comment]
If I use the default step of 1h, a value of `07:40:25` is displayed as `07:40`, so the seconds are missing.

**Comment by rolfsmeds** [comment]
That's true: although the step size does not prevent you from entering a value with higher precision within the same order of magnitude, it does affect the supported time _format_, and so prevents you from entering a value with a smaller unit (except if the step is hours, in which case you can still enter minutes) so in order to be able to enter seconds, the step must be less than 1min, and to enter fractions it must be less than 1sec.

The javadoc defines this behavior as follows:
> If the step is less than 60 seconds, the format will be changed to hh:mm:ss and it can be in hh:mm:ss.fff format, when the step is less than 1 second.

So what I think would make sense is to provide an API for setting the _format_ separately from the step (somewhat analogously to https://github.com/vaadin/vaadin-date-picker/issues/761)

I'll change the ticket title accordingly, if you don't mind.
